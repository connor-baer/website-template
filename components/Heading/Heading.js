import styled from '@emotion/styled';
import { css } from '@emotion/core';

const sizeMap = {
  h1: {
    fontSize: ['2.25rem', '2.75rem']
  },
  h2: {
    fontSize: ['1.75rem', '2rem']
  },
  h3: {
    fontSize: ['1.33rem', '1.5rem']
  },
  h4: {
    fontSize: ['1.25rem', '1.25rem']
  }
};

const baseStyles = ({ theme, size, as = 'h2' }) => {
  const { fontSize } = sizeMap[size || as];

  return css`
    font-size: ${fontSize[0]};
    line-height: 1.5;
    letter-spacing: 0.15px;
    text-decoration: none;
    margin: 1em 0 0;

    ${theme.mq.kilo} {
      font-size: ${fontSize[1]};
    }

    a {
      text-decoration: none;

      &:hover,
      &:focus {
        text-decoration: underline;
      }
    }
  `;
};

const Heading = styled('h2')(baseStyles);

export default Heading;
