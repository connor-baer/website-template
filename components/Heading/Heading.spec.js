import React from 'react';

import Heading from './Heading';

describe('Heading', () => {
  function renderHeading(fn, props = {}) {
    return fn(<Heading {...props} />);
  }

  it('should render with default styles', () => {
    const children = 'Lychee';
    const { container } = renderHeading(render, { children });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with h1 styles', () => {
    const children = 'Lychee';
    const { container } = renderHeading(render, { children, as: 'h1' });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with h3 styles', () => {
    const children = 'Lychee';
    const { container } = renderHeading(render, { children, as: 'h3' });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with h4 styles', () => {
    const children = 'Lychee';
    const { container } = renderHeading(render, { children, as: 'h4' });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const children = 'Lychee';
    const html = renderHeading(renderToHtml, { children });
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
