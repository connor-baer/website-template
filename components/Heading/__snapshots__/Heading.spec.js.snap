// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render with default styles 1`] = `
.circuit-0 {
  font-size: 1.75rem;
  line-height: 1.5;
  -webkit-letter-spacing: 0.15px;
  -moz-letter-spacing: 0.15px;
  -ms-letter-spacing: 0.15px;
  letter-spacing: 0.15px;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 1em 0 0;
}

@media (min-width:600px) {
  .circuit-0 {
    font-size: 2rem;
  }
}

.circuit-0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.circuit-0 a:hover,
.circuit-0 a:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<h2
  class="circuit-0 circuit-1"
>
  Lychee
</h2>
`;

exports[`Heading should render with h1 styles 1`] = `
.circuit-0 {
  font-size: 2.25rem;
  line-height: 1.5;
  -webkit-letter-spacing: 0.15px;
  -moz-letter-spacing: 0.15px;
  -ms-letter-spacing: 0.15px;
  letter-spacing: 0.15px;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 1em 0 0;
}

@media (min-width:600px) {
  .circuit-0 {
    font-size: 2.75rem;
  }
}

.circuit-0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.circuit-0 a:hover,
.circuit-0 a:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<h1
  class="circuit-0 circuit-1"
>
  Lychee
</h1>
`;

exports[`Heading should render with h3 styles 1`] = `
.circuit-0 {
  font-size: 1.33rem;
  line-height: 1.5;
  -webkit-letter-spacing: 0.15px;
  -moz-letter-spacing: 0.15px;
  -ms-letter-spacing: 0.15px;
  letter-spacing: 0.15px;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 1em 0 0;
}

@media (min-width:600px) {
  .circuit-0 {
    font-size: 1.5rem;
  }
}

.circuit-0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.circuit-0 a:hover,
.circuit-0 a:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<h3
  class="circuit-0 circuit-1"
>
  Lychee
</h3>
`;

exports[`Heading should render with h4 styles 1`] = `
.circuit-0 {
  font-size: 1.25rem;
  line-height: 1.5;
  -webkit-letter-spacing: 0.15px;
  -moz-letter-spacing: 0.15px;
  -ms-letter-spacing: 0.15px;
  letter-spacing: 0.15px;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 1em 0 0;
}

@media (min-width:600px) {
  .circuit-0 {
    font-size: 1.25rem;
  }
}

.circuit-0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.circuit-0 a:hover,
.circuit-0 a:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<h4
  class="circuit-0 circuit-1"
>
  Lychee
</h4>
`;
